src/easydmp/plan/views/__init__.py

```diff
@@ -745,25 +758,21 @@ def get_queryset(self):
             qs = qs | self.model.objects.viewable(self.request.user)
         return qs.distinct()
 
-    def get_context_data(self, **kwargs):
-        context = self.object.get_context_for_generated_text()
-        context.update(**kwargs)
-        context['logs'] = EventLog.objects.any(self.object)
-        context['reveal_questions'] = self.object.template.reveal_questions
-        context['editors'] = ', '.join([str(ed) for ed in self.get_editors()])
-        context['last_validated_ok'] = self.object.last_validated if self.object.valid else '-'
-        return super().get_context_data(**context)
+    def generate_exported_plan(self):
+        self.object = self.get_object()
+        template = self.get_template_names()[0]
+        self.export = generate_pretty_exported_plan(self.object, template)
 
-    def get(self, request, *args, **kwargs):
-        next = super().get(request, *args, **kwargs)
+    def log(self, request):
         template = '{timestamp} {actor} accessed generated version of {target}'
         log_event(request.user, 'access generated plan', target=self.object,
                   template=template)
-        return next
 
-    def get_editors(self):
-        return [access.user for access in
-                PlanAccess.objects.filter(plan=self.object).filter(may_edit=True).all()]
+    def get(self, request, *args, **kwargs):
+        self.generate_exported_plan()
```

Comment:
Well, do we need to store the export on self at all here? Isn't it cleaner to keep it function local?

Reply:
That method is strictly doing too much anyway but I wanted to keep `get` short and DRY. I had `log` in there for a while too.

I think I'll keep it as is, but the underlying wart is how get/post etc. works in CBVs at all, it mixes up too much stuff. What if you want to do something with the form-data before you clean, for instance.